sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/state/StatePartitionReader.scala

```diff
@@ -256,4 +271,21 @@ class StateStoreChangeDataPartitionReader(
     result.update(4, partition.partition)
     result
   }
+
+  private def createFlattenedRow(
```

Comment:
Is the method expected to be used except MapState? Otherwise I'd just name the method less general / more specific to MapState.

Reply:
Done